class Solution {
    public void rotate(int[][] matrix) {
        int n = matrix.length;
        int[][] arr = new int[n][n];
        for(int i=0; i<=n-1; i++){
            for(int j=0; j<=n-1; j++){
                arr[j][n-1-i] = matrix[i][j];
            }
        }
        //return arr into matrix
        for(int i=0; i<=n-1; i++){
            for(int j=0; j<=n-1; j++){
                matrix[i][j] = arr[i][j];
            }
        }
    }
}